numba/_helperlib.c

```diff
@@ -563,7 +563,42 @@ import_cython_function(const char *module_name, const char *function_name)
     }
     /* 2.7+ => Cython exports a PyCapsule */
     capsule_name = PyCapsule_GetName(cobj);
-    if (capsule_name != NULL) {
+    if (PyCapsule_IsValid(cobj,capsule_name)!=0) {
```

Comment:
```suggestion
    if (PyCapsule_IsValid(cobj, capsule_name) != 0) {
```
also, should this raise if invalid?